File: lint-arwaky/archive/contract/config_validation_port.py
"""config_validation_port — Interface for configuration validation and orchestration."""

from abc import ABC, abstractmethod
from ..taxonomy import FilePath, AppConfig, ConfigError


class IConfigValidatorPort(ABC):
    """Port for validating and orchestrating application configuration."""

    @abstractmethod
    def load_config(
        self,
        env_path: FilePath | None = None,
        yaml_path: FilePath | None = None,
    ) -> AppConfig | ConfigError:
        """Load or reload configuration. Returns AppConfig."""
        ...

    @abstractmethod
    def get_config(self) -> AppConfig | ConfigError:
        """Get the current configuration."""
        ...

    @abstractmethod
    def reset_config(self) -> None:
        """Reset the configuration."""
        ...

File: lint-arwaky/src-rust/contract/config_validation_port.rs
// config_validation_port — Interface for configuration validation and orchestration.
use crate::taxonomy::{AppConfig, ConfigError, FilePath};
use async_trait::async_trait;

#[async_trait]
pub trait IConfigValidationPort: Send + Sync {
    /// Load or reload configuration. Returns AppConfig.
    async fn load_config(
        &self,
        env_path: Option<FilePath>,
        yaml_path: Option<FilePath>,
    ) -> Result<AppConfig, ConfigError>;

    /// Get the current configuration.
    async fn get_config(&self) -> Result<AppConfig, ConfigError>;

    /// Reset the configuration.
    async fn reset_config(&self);
}
